lib/clearance/back_door.rb

```diff
@@ -31,8 +31,8 @@ module Clearance
   #   visit new_feedback_path(as: user)
   class BackDoor
     def initialize(app, &block)
-      unless ENV["RAILS_ENV"] == "test"
-        raise "Can't use backdoor outside test environment"
+      unless Clearance.configuration.allowed_backdoor_environments.include? ENV["RAILS_ENV"]
```

Comment:
I realize it wasn't like this before you changed - but can we extract this to private method?

Reply:
Are you asking to make initialize a private method or the unless check a private method?

Additionally for the comment are you asking me to add extra unit tests to check different envs?

Reply:
Private method within `BackDoor` for the check - something like `unless allowed_backdoor_env?` or similar.

And yes - a unit test that shows the configuration overriding the default set of envs.

Reply:
I see I'll make a private method to check the environment and call it instead.

I'm not too familiar with rspec though I came up with this based on the other tests? Would this work for our needs?
```ruby
it "can be used with non-default explicitly allowed envs" do
    Clearance.configuration.allowed_backdoor_environments.push("demo")
    with_environment("RAILS_ENV" => "demo") do
      user_id = "123"
      user = double("user")
      allow(User).to receive(:find).with(user_id).and_return(user)
      env = env_for_user_id(user_id)
      back_door = Clearance::BackDoor.new(mock_app)

      result = back_door.call(env)

      expect(env[:clearance]).to have_received(:sign_in).with(user)
      expect(result).to eq mock_app.call(env)
    end 
  end
```